src/strategies/erc20-received/index.ts

```diff
@@ -0,0 +1,96 @@
+import fetch from 'cross-fetch';
+import { Web3Provider } from '@ethersproject/providers';
+import { formatUnits } from '@ethersproject/units';
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+
+export const author = 'mccallofthewild';
+export const version = '0.1.0';
+
+export async function strategy(
+  ...args: [
+    string,
+    string,
+    Web3Provider,
+    string[],
+    {
+      coeff?: number;
+      receivingAddresses: string[];
+      contractAddress: string;
+      decimals: number;
+    },
+    number
+  ]
+) {
+  const [space, network, provider, addresses, options, snapshot] = args;
+  const { coeff = 1, receivingAddresses, contractAddress, decimals } = options;
+
+  const etherscanApiKey = 'TRBDWD7I5UK25G1ZZAZ3I889S6F4KBP4BN';
+  const logResults: {
+    fromAddress: string;
+    toAddress: string;
+    amount: number;
+  }[] = [];
+  const requestDelay = 1000 / 5;
+  let iteration = 0;
+  let lastFetchTime = Date.now();
+
+  for (let fromAddress of addresses)
+    for (let toAddress of receivingAddresses) {
+      await new Promise((r) =>
+        setTimeout(r, requestDelay - Math.max(0, Date.now() - lastFetchTime))
+      );
+      const r = await fetch(
```

Comment:
The issue is that this type of request don't scale well, there is some proposal with 1000 votes but we don't want user send 1000 request in their browser. Is there a different API you could use to achieve this?

Reply:
It turns out that web3's `#getLogs` method is perfect for this. Just pushed the update 👍

Reply:
Hum i think with getLogs there is a limit of 1000 results, can you try with an address with long history to see if that break or not?

Reply:
Dfuse (https://docs.dfuse.io) could be an alternative solution

Reply:
Infura has an increased log limit of 10,000 results, but Dfuse is definitely the way to go. It's more consistent and *extremely* scalable.  That was a great suggestion